samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemAdmin.java

```diff
@@ -231,11 +231,13 @@ public String getUpcomingOffset() {
             streamNames.forEach(streamName -> {
               Map<Partition, SystemStreamMetadata.SystemStreamPartitionMetadata> partitionMetadata = new HashMap<>();
 
-              List<PartitionInfo> partitionInfos = metadataConsumer.partitionsFor(streamName);
-              LOG.debug("Stream {} has partitions {}", streamName, partitionInfos);
-
-              partitionInfos.forEach(partitionInfo -> partitionMetadata.put(new Partition(partitionInfo.partition()), dummySspm));
+              synchronized (metadataConsumer) {
```

Comment:
Let's add a comment on the metadataConsumer field that all access needs to be synchronized (and explain why). 

Also, if possible let's make it private.